lib/CL/pocl_binary.c

```diff
@@ -0,0 +1,615 @@
+/* OpenCL runtime library: pocl binary
+
+   Copyright (c) 2016 pocl developers
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+*/
+
+#include <stdint.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "pocl_cl.h"
+#include "pocl_binary.h"
+#include "pocl_cache.h"
+#include "pocl_file_util.h"
+
+#include <sys/stat.h>
+#include <dirent.h>
+#include <libgen.h>
+
+#if defined(WORDS_BIGENDIAN) && WORDS_BIGENDIAN == 1
+  const char host_endian = 1;
+#else
+  const char host_endian = 0;
+#endif
+
+/* pocl binary identifier */
+#define POCLCC_STRING_ID "poclbin"
+#define POCLCC_STRING_ID_LENGTH 7
+#define POCLCC_VERSION 1
+
+/* pocl binary structures */
+
+typedef struct pocl_binary_kernel_s {
+  /* the first 3 fields are sizes in bytes of the data pieces that follow
+   * (to allow fast skipping) */
+
+  // size of the entire struct
+  uint64_t struct_size;
+  // size of tared content
```

Comment:
Please use something else than 'tar' here as it's not tar (it's a well known archive format) but even simpler.